pennylane/hf/tapering.py

```diff
@@ -464,3 +467,106 @@ def transform_hamiltonian(h, generators, paulix_ops, paulix_sector):
     c = qml.math.stack(c)
 
     return _simplify(qml.Hamiltonian(c, o))
+
+
+def _sector_energy(sector):
+    """Computes energy of the given tapered Hamiltonian.
+
+    Args:
+        sector (pennylane.Hamiltonian): tapered Hamiltonian whose energy has to be calculated
+
+    Returns:
+        energy (float): energy of the tapered hamiltonian
+    """
+
+    if len(sector.wires) < 2:
+        energy = min(sp.linalg.eigvals(qml.utils.sparse_hamiltonian(sector).toarray()))
+    else:
+        energy = sp.sparse.linalg.eigs(qml.utils.sparse_hamiltonian(sector), k=2)[0].min()
+
+    return energy
+
+
+def find_optimal_sector(qubit_op, generators, paulix_ops, brute_force=True, num_electrons=0):
```

Comment:
I also recommend renaming to `optimal_sector`. As a general rule, I would like function names to have at most one underscore `_`. 

In fact, we likely will need a big refactor of the function names in the HF module since right now half of them start with `generate`

Reply:
I have renamed the function now `optimal_sector`.